scheduler/src/cook/mesos/rebalancer.clj

```diff
@@ -262,7 +262,7 @@
         (reduce (fn [task-ents-by-user task-ent]
                   (let [user (util/task-ent->user task-ent)
                         f (if (= new-running-task-ent task-ent)
-                            (fnil conj (sorted-set-by (util/same-user-task-comparator-penalize-backfill)))
+                            (fnil conj (sorted-set-by (util/same-user-task-comparator)))
```

Comment:
Ah, got it! For some reason I (incorrectly) felt that `user` was one of the keys in `user->sorted-running-task-ents`. 

The `update-in` below is either initializing the entry against a user to a sorted set or adding elements into the `sorted-running-task-ents'` set.